src/player.js

```diff
@@ -1849,8 +1847,10 @@ export default class Player extends FakeEventTarget {
    */
   _updateTracks(tracks: Array<Track>): void {
     Player._logger.debug('Tracks changed', tracks);
-    this._tracks = [...tracks, ...this._externalCaptionsHandler.getExternalTracks(tracks)];
+    this._tracks = tracks.concat(this._externalCaptionsHandler.getExternalTracks(tracks));
     this._addTextTrackOffOption();
+    this._maybeSetTracksLabels();
```

Comment:
why move this inside the method and not keep it as before? because 608/708 can update on the fly and you want to call `_updateTracks` and apply `_maybeSetTracksLabels` and `_setDefaultTracks`.
You need to check the `_setDefaultTracks` cause you only want it to happen once and in this case if there's some unwanted side effect it might be called several times, when ever `_onTracksChanged` is called.

Reply:
I've checked this change of course.
it behaves correct thanks `this._playbackAttributesState.textLanguage`